QModManager/Utility/Dialog.cs

```diff
@@ -71,47 +71,70 @@ internal enum DialogColor
         internal Button rightButton = Button.Disabled;
 
         private GameObject coroutineHandler;
+        private float pos = 0;
+        private Sprite Sprite;
 
         internal void Show()
         {
-            coroutineHandler = new GameObject("QModManager Dialog Coroutine");
-            coroutineHandler.AddComponent<DummyBehaviour>().StartCoroutine(ShowDialogEnumerator());
+            if(coroutineHandler is null)
+            {
+                coroutineHandler = new GameObject("QModManager Dialog Coroutine");
+                coroutineHandler.AddComponent<DummyBehaviour>().StartCoroutine(ShowDialogEnumerator());
+            }
+            else if(coroutineHandler.TryGetComponent(out DummyBehaviour dummyBehaviour))
+            {
+                dummyBehaviour.StartCoroutine(ShowDialogEnumerator());
+            }
+            else
+            {
+                coroutineHandler.AddComponent<DummyBehaviour>().StartCoroutine(ShowDialogEnumerator());
+            }
```

Comment:
Could this work instead? _replacing lines 79-91_
```CS
if(coroutineHandler is null)
{
    coroutineHandler = new GameObject("QModManager Dialog Coroutine");
}
var dummyBehaviour = coroutineHandler.EnsureComponent<DummyBehaviour>();
dummyBehaviour.StartCoroutine(ShowDialogEnumerator());
```

Reply:
yea it should.